addons/delivery/models/stock_picking.py

```diff
@@ -190,6 +224,20 @@ def check_packages_are_identical(self):
                 raise UserError(_('You are shipping different packaging types in the same shipment.\nPackaging Types: %s' % package_names))
         return True
 
+    @api.multi
+    @api.returns('self', lambda value: value.id)
+    def message_post(self, **kwargs):
+        """ Overrides mail_thread message_post so that we can set delivery_mail_sent to true  of boolean field value to identify email 
+        is sent and make send confirmation email button normal from primary.
+        """
+        if self.env.context.get('mark_delivery_as_sent'):
+            for picking in self:
+                if picking.state == 'done':
+                    picking.delivery_mail_sent = True
+        return super(StockPicking, self).message_post(**kwargs)
```

Comment:
Yes email posted in chatter. but 
actual problem (why we need to override this ) is once mail is sent we have to make 'send confirmation email' button normal instead of primary. for that we have make one boolean field like flag and updated when compose send mail called.( that was kind of hack already done on lot of modules) but after discussion with yti and tde we have created task for that 
see this : 
https://www.odoo.com/web#id=32721&view_type=form&model=project.task&menu_id=4720

and instead of send_mail we override message_post.